SYCL/Basic/accessor/accessor.cpp

```diff
@@ -683,6 +683,75 @@ int main() {
     }
   }
 
+  // placeholder accessor exception (2) // SYCL2020 4.7.6.9
+  {
+    sycl::queue q;
+    // host device executes kernels via a different method and there
+    // is no good way to throw an exception at this time.
+    sycl::range<1> r(4);
+    sycl::buffer<int, 1> b(r);
+    try {
+      using AccT = sycl::accessor<int, 1, sycl::access::mode::read_write,
+                                  sycl::access::target::device,
+                                  sycl::access::placeholder::true_t>;
+      AccT acc(b);
+
+      q.submit([&](sycl::handler &cgh) {
+        // we do NOT call .require(acc) without which we should throw a
+        // synchronous exception with errc::kernel_argument
+        // The difference with the previous test is that the use of acc
+        // is usually optimized away for this particular scenario, but the
+        // exception should be thrown because of passing it, not because of
+        // using it
+        cgh.single_task<class ph>([=] { int x = acc[0]; });
+      });
+      q.wait_and_throw();
+      assert(false && "we should not be here, missing exception");
+    } catch (sycl::exception &e) {
+      std::cout << "exception received: " << e.what() << std::endl;
+      assert(e.code() == sycl::errc::kernel_argument && "incorrect error code");
+    } catch (...) {
+      std::cout << "some other exception" << std::endl;
```

Comment:
Nit; Maybe include `__LINE__` in the printed message just so it is easier to find our way to which case failed.